core/templates/services/improvements-backend-api.service.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for fetching improvement tasks from the backend.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { ExplorationTask, ExplorationTaskObjectFactory } from
+  'domain/improvements/ExplorationTaskObjectFactory';
+import { ITaskEntryBackendDict } from
+  'domain/improvements/TaskEntryObjectFactory';
+import { ImprovementsConstants } from
+  'domain/improvements/improvements.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+export interface IExplorationImprovementsResponseBackendDict {
+  'open_tasks': ITaskEntryBackendDict[];
+  'resolved_task_types_by_state_name': {
+    [stateName: string]: string[];
+  };
+}
+
+export interface IExplorationImprovementsHistoryResponseBackendDict {
+  'results': ITaskEntryBackendDict[];
+  'cursor': string;
+  'more': boolean;
+}
+
+export class ExplorationImprovementsResponse {
+  constructor(
+      public readonly openTasks: ExplorationTask[],
+      public readonly resolvedTaskTypesByStateName: Map<string, string[]>) {}
+}
+
+export class ExplorationImprovementsHistoryResponse {
+  constructor(
+      public readonly results: ExplorationTask[],
+      public readonly cursor: string,
+      public readonly more: boolean) {}
+}
+
+@Injectable({providedIn: 'root'})
+export class ImprovementsBackendApiService {
+  constructor(
+      private explorationTaskObjectFactory: ExplorationTaskObjectFactory,
+      private http: HttpClient,
+      private urlInterpolationService: UrlInterpolationService) {}
+
+  async getTasksAsync(expId: string): Promise<ExplorationImprovementsResponse> {
+    const explorationImprovementsUrl = (
+      this.urlInterpolationService.interpolateUrl(
+        ImprovementsConstants.EXPLORATION_IMPROVEMENTS_URL, {
+          exploration_id: expId
+        }));
+    return this.http.get<IExplorationImprovementsResponseBackendDict>(
+      explorationImprovementsUrl
+    ).toPromise().then(
+      backendDict => new ExplorationImprovementsResponse(
+        backendDict.open_tasks
+          .map(d => this.explorationTaskObjectFactory.createFromBackendDict(d)),
```

Comment:
Optional: Consider breaking after '(' instead of before the '.' -- it's a bit more readable, I think.

Reply:
Hm, I think I prefer the other one, but it's up to you. The thing is, the previous line, taken on its own, doesn't really suggest that a continuation line is present, which the open paren does -- so I think the open paren line break is a bit easier to read. But it's your call.

Reply:
Acknowledged